fem/restriction.hpp

```diff
@@ -162,6 +162,18 @@ class FaceRestriction : public Operator
    */
    virtual void AddMultTranspose(const Vector &x, Vector &y) const = 0;
 
+   /** @brief Add the face degrees of freedom @a x to the element degrees of
+       freedom @a y.
+
+       @param[in]     x The face degrees of freedom on the face.
+       @param[in,out] y The L-vector of degrees of freedom to which we add the
+                        face degrees of freedom.
+   */
+   virtual void AddMultTranspose(Vector &x, Vector &y) const
+   {
+      AddMultTranspose((const Vector &)x, y);
```

Comment:
What is the motivation behind this cast?

Reply:
Using the non-destructive version of the function, I will rename the function instead to make this more explicit.

Reply:
I've added `InPlace` to methods doing "in place" computation in [d31979c](https://github.com/mfem/mfem/pull/3105/commits/d31979c02ee6b057af6b283bd01c7f731a9756fd).